packages/react/src/primitives/Select/Select.tsx

```diff
@@ -23,11 +24,16 @@ export const Select: React.FC<SelectProps> = (props) => {
     ...rest
   } = props;
   return (
-    <View className="amplify-select-wrapper">
+    <View className={ComponentClassNames.SelectWrapper}>
       <View
+        aria-invalid={hasError}
         as="select"
         value={value}
-        defaultValue={defaultValue}
+        defaultValue={
+          value === undefined && defaultValue === undefined && placeholder
```

Comment:
value === undefined is to get rid of controlled component. React will complain if we set both defaultValue and value

Reply:
> Should we check for `null` and `undefined` using loose equality check? (`== null` will check for `null `and `undefined`)

React will take `null` as a valid value passed to a prop. It just indicates nothing. So we will got the same complain if we rule it out along with `undefined`. Also, it will warn when `value` is set to `null` and request us to instead use empty string or `undefined`.

Reply:
Yeah, that's a good point. if it's set to null then the customer passed it in, versus undefined means the prop wasn't given. 👍